plugins/main/public/react-services/navigation-service.tsx

```diff
@@ -124,6 +124,44 @@ class NavigationService {
     return getCore().application.getUrlForApp(appId, options);
   }
 
+  public buildSearch(search: URLSearchParams) {
+    return Array.from(search.entries())
+      .map(([key, value]) => `${key}=${value}`)
+      .join('&');
+  }
+
+  public updateAndNavigateSearchParams(
+    updateParams: { [key: string]: string | undefined },
+    deleteParams: string[] = [],
```

Comment:
**suggestion**: you could simplify the logic using only a parameter of object type. If the value of a key is `null`, then the parameter could be considered to be removed.

Example of usage:

```
navigationService.updateAndNavigateSearchParams({
  param1: null,
});
```

Reply:
This has been done in this commit [363c107](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6775/commits/363c10731ff2d0de23ecfbcdd013cc2bb072280c)